cranelift/codegen/src/machinst/abi.rs

```diff
@@ -608,26 +608,26 @@ pub trait ABIMachineSpec {
 pub struct Sig(u32);
 cranelift_entity::entity_impl!(Sig);
 
+impl Sig {
+    fn prev(self) -> Option<Sig> {
+        self.0.checked_sub(1).map(|x| Sig(x))
```

Comment:
This isn't important, but I personally prefer to not use a closure if it's just passing its argument to a function or a tuple constructor. Like this:

```suggestion
        self.0.checked_sub(1).map(Sig)
```